webapp/src/main/resources/public/js/components/users/UserModal.js

```diff
@@ -28,6 +28,71 @@ class UserModal extends React.Component{
         return 'success';
     }
 
+    renderLocales() {
+        const options = [];
+        for (let tag of this.props.locales.allLocales.map((x) => x.bcp47Tag).toSorted()) {
+            if (this.props.modal.localeTags.includes(tag) || (this.props.modal.localeFilter && !tag.toLowerCase().includes(this.props.modal.localeFilter.toLowerCase()))) {
+                continue;
+            }
+            options.push(
+                <MenuItem key={tag} eventKey={tag} active={tag === this.props.modal.selectedLocale}>
+                    {tag}
+                </MenuItem>
+            );
+        }
+
+        let localeElements = [];
+        for (let tag of this.props.modal.localeTags.toSorted()) {
+            localeElements.push(
+                <Button key={tag + "-btn"} bsStyle="danger" onClick={() => UserModalActions.removeLocaleFromList(tag)}>
+                    <span className="glyphicon glyphicon-trash foreWhite" aria-label={"remove locale " + tag}/>
+                </Button>
+            );
+            localeElements.push(<span key={tag + "-display"}>{tag}</span>);
+        }
+
+        return (
+            <div className="mtm users-locale-select-root">
+                <div className="locales-list panel panel-default">
+                    {localeElements}
+                </div>
+                <Dropdown
+                    id="all-locales"
+                    style={{gridArea: 'new-select'}}
+                >
+                    <Dropdown.Toggle
+                        id="all-locales-btn"
+                        style={{width: '100%'}}
+                        onClick={() => UserModalActions.updateLocaleFilter('')}
+                    >
+                        {this.props.modal.selectedLocale}
+                    </Dropdown.Toggle>
+                    <Dropdown.Menu
```

Comment:
With the a dropdown, I would be against it, since the default expectation with a dropdown element is that all elements are present, and not that typing to find a specific locale is mandatory.

I could change it to a regular input field and use a [Datalist](https://getbootstrap.com/docs/5.3/forms/form-control/#datalists) instead, but then you would loose the option to browse through all available locales.

Reply:
My feeling is that if someone is selecting locales that a user is allowed to translate then they'll likely be coming with a short list of specific locales to enable so I would be leaning towards a Datalist and dropping the ability to scroll through all locales in a dropdown. 

@aurambaj any thoughts on this?

Reply:
Changed it to a datalist for now.